python/taichi/ui/window.py

```diff
@@ -16,12 +16,13 @@ class Window(_ti_core.PyWindow):
         res (Tuple[Int]): resolution (width, height) of the window, in pixels.
         layout (vsync): whether or not vertical sync should be enabled.
     """
-    def __init__(self, name, res, vsync=False, show_window=True):
+    def __init__(self, name, res, vsync=False, show_window=True, prog=None):
```

Comment:
Thanks! I think it's fine to use `get_runtime().prog` at Python level. The CPP part needs to get rid of these global states.